compiler/optimizer/parallelize.go

```diff
@@ -253,7 +253,7 @@ func (o *Optimizer) concurrentPath(ops []dag.Op, sortKey order.SortKey) (int, or
 				return 0, order.Nil, false, false, nil
 			}
 			return k, newKey, false, true, nil
-		case *dag.Parallel, *dag.Head, *dag.Tail, *dag.Uniq, *dag.Fuse, *dag.Sequential, *dag.Join:
+		case *dag.Parallel, *dag.Head, *dag.Tail, *dag.Uniq, *dag.Fuse, *dag.Sequential, *dag.Join, *dag.Load:
```

Comment:
dag.Load wants ordered input (hence `true, true`) but doesn't produce ordered output (hence `order.Nil`).
```suggestion
		case *dag.Load:
			return k, order.Nil, true, true, nil
		case *dag.Parallel, *dag.Head, *dag.Tail, *dag.Uniq, *dag.Fuse, *dag.Sequential, *dag.Join:
```